File: apple/Sources/ReactNativeWatchOS/Modifiers/FormStyle.swift
import SwiftUI

/// SwiftUI `.formStyle(_:)`. watchOS ships only `AutomaticFormStyle`
/// (`.automatic`); `.columns` (macOS) and `.grouped` (iOS/macOS) are
/// unavailable on watchOS, so every value maps to `.automatic`. The `style`
/// param is accepted for cross-platform parity but has no other effect here.
enum RNWFormStyleModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("formStyle") { view, _, _ in
            AnyView(view.formStyle(.automatic))
        }
    }
}

File: src/modifiers/formStyle.ts
import { createModifier } from './createModifier';

export type FormStyle = 'automatic' | 'columns' | 'grouped';

/// SwiftUI `.formStyle(_:)`. Sets the visual style applied to `Form`s
/// within the view (watchOS 9+). `'automatic'` defers to the platform
/// default.
export function formStyle(style: FormStyle = 'automatic') {
  return createModifier('formStyle', { style });
}
